docs/design/908-proposal-process.md

```diff
@@ -0,0 +1,410 @@
+documented and the proposal will move to the appropriate state.
+
+If the governance team cannot reach consensus, the path forward is called to a
+formal governance [vote][] as a last resort. The result of the vote is
+documented and the proposal will move to the appropriate state.
+
+[governance]: https://github.com/grafana/alloy/blob/main/GOVERNANCE.md
+[vote]: https://github.com/grafana/alloy/blob/main/GOVERNANCE.md#voting
+
+## Rationale
+
+Formalizing proposal creation and review ensures that all proposals are
+eventually given attention. Having proposals reviewed weekly ensures that
+proposals are kept moving forward and nobody is left waiting months for a
+decision to be made.
+
+To re-embrace the spirit of "design in the open," conversations about a
+proposal should be discoverable by new parties to make it easy to participate.
+Having a single source of truth for discussions (the proposal issue or design
+document PR) prevents concurrent conversations from fragmenting in multiple
+locations. This ensures that contributors know where to look for active
+discussions and important information doesn't get lost while trying to reach
+consensus.
+
+However, this process comes with some trade-offs, described in sections below.
+
+### Proposal lifecycle time
+
+Formalizing the proposal process ensures that all proposals are given
+attention. However, the process described here will slow down the acceptance
+rate of proposals.
+
+Because changing a proposal's state is tied to the frequency of proposal review
+process, a weekly proposal review process means that the acceptance cycle of
+most proposals is increased to take at least two or three weeks; up to one week
+to move to _Active_, at least one week to move to _Likely Accept_, and at least
+one week to move to _Accepted_. This does not include the time a proposal may
+spend in _Hold_.
+
+Three weeks is half the time of our release cadence, and so proposals opened
+too late in the cycle may need to wait two releases before being available to
+users.
+
+However, if proposals are needed more urgently, the proposal review group can
+fast-track proposals. This should mitigate the impact of this trade-off. If
+this mitigation is not sufficient, further proposals can be explored to reduce
+the time it takes to get a proposal approved and implemented into a release.
+
+### Conversational toil
+
+Using GitHub for all proposals ensures there is a single source of truth at a
+given time for discussion, and that all interested parties can participate in
+shaping the future of Alloy.
+
+However, GitHub is not well-suited for long-form discussions. This may lead to
+long threads that can be difficult to follow.
+
+There are some mitigation strategies to help with this:
+
+* Discussion on issues should be done conversationally rather than responding
+  to individual lines of text. This can help reduce the total number of
+  comments and make a conversation more digestible.
```

Comment:
As mentioned before, I find long-form essays by far less digestible than comments on individual lines of text, where all the context is available. I feel the argument in your doc is up for the debate and could actually have quite the opposite effect. A lot (if not all) modern communication tools for large groups of people support threads, so my view on this topic is also likely to be quite common.

Having said that, I'm happy that PR with design doc can become the source of truth for a more involved discussion. In fact, I would consider a policy where anything more than two threads or two rounds of technical comments should usually be migrated to a design doc PR. Worst case it'd be a short design doc.